src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -420,17 +424,30 @@ void updateGeneratorsState(double generationQ, boolean reactiveLimits) {
     }
 
     @Override
-    public void updateState(boolean reactiveLimits, boolean writeSlackBus) {
+    public void updateState(boolean reactiveLimits, boolean writeSlackBus, boolean distributedOnConformLoad, boolean loadPowerFactorConstant) {
         // update generator reactive power
         updateGeneratorsState(voltageControllerEnabled ? calculatedQ + loadTargetQ : generationTargetQ, reactiveLimits);
 
         // update load power
-        double factorP = initialLoadTargetP != 0 ? loadTargetP / initialLoadTargetP : 1;
-        double factorQ = initialLoadTargetQ != 0 ? loadTargetQ / initialLoadTargetQ : 1;
+        double diffTargetP = loadCount > 0 ? loadTargetP - initialLoadTargetP : 0;
+        double sumUpdatedQ = 0;
         for (Load load : loads) {
+            double updatedP0 = load.getP0() + diffTargetP * Math.abs(load.getP0()) / absLoadTargetP;
+            double updateQ0 = load.getQ0();
+            if (distributedOnConformLoad) {
+                updatedP0 = load.getExtension(LoadDetail.class) == null ? load.getP0() :
+                        load.getP0() + diffTargetP * Math.abs(load.getExtension(LoadDetail.class).getVariableActivePower()) / absVariableLoadTargetP;
+            }
+            if (loadPowerFactorConstant) {
+                updateQ0 = load.getQ0() / load.getP0() * updatedP0;
+            }
             load.getTerminal()
-                    .setP(load.getP0() >= 0 ? factorP * load.getP0() : load.getP0())
-                    .setQ(load.getQ0() >= 0 ? factorQ * load.getQ0() : load.getQ0());
+                    .setP(updatedP0)
+                    .setQ(updateQ0);
+            sumUpdatedQ += updateQ0;
+        }
+        if (Math.abs(sumUpdatedQ - loadTargetQ) > Q_DISPATCH_EPSILON) {
+            LOGGER.error("FIXME"); // it happens when we have a load with negative P0 in the bus.
```

Comment:
We cannot let this as it is because when running on some cases, the output is cluttered with FIXME.
Do we know how to fix it?

Reply:
No, I really don't know how to fix that. Keeping power factor constant for the whole bus works only if all loads have positive set points, otherwise I have to think. For sure, it cannot let a `FIXME` here, but I have no idea of how to handle that.

Reply:
I have solve this issue by introducing `LfLoad` new elements. It works and I really don't know how to do this without this new object. Just take a look and tell me your opinion.